src/share-buttons.js

```diff
@@ -145,13 +156,29 @@
             var handler = function () {
                 share(opt.id, opt.url, opt.title, opt.desc);
             };
+            var iehandler = function () {
+                handler.call(el);
+            };
+            var cachedRef = el.getAttribute('data-sharebtn-ref')
```

Comment:
Great idea to store refs, but I think it will be more clear when this logic will be split as init (add listeners) and deinit/dispose (remove listeners, clear states), after that update() can compose these methods

Reply:
Yeah, but actually update use `this.i()` to *reinit*  `.share-btn` expected data and actions. Splitting the process with a `deinit` function address the challenge to loop links `a` twice to add and remove listeners. However it can be more convenient to process `removeEventListener` and then `addEventListener`  in `prepareLink` step. This like you suggest, to be more clear on that logic.